pkg/messaging/mqtt/pubsub_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package mqtt_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/mainflux/mainflux/pkg/messaging"
+	"github.com/mainflux/mainflux/pkg/messaging/mqtt"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	topic       = "topic"
+	chansPrefix = "channels"
+	channel     = "9b7b1b3f-b1b0-46a8-a717-b8213f9eda3b"
+	subtopic    = "engine"
+	clientID    = "9b7b1b3f-b1b0-46a8-a717-b8213f9eda3b"
+)
+
+var (
+	msgChan = make(chan messaging.Message)
+	data    = []byte("payload")
+)
+
+func TestPublisher(t *testing.T) {
+	err := pubsub.Subscribe(clientID, fmt.Sprintf("%s.%s", chansPrefix, topic), handler{})
+	require.Nil(t, err, fmt.Sprintf("got unexpected error: %s", err))
+	err = pubsub.Subscribe(clientID, fmt.Sprintf("%s.%s.%s.", chansPrefix, topic, subtopic), handler{})
```

Comment:
Why are you subscribing 2 times in a row? Are you testing something? If yes, explain what you test in a comment (while other tests below have their `desc`, here it is not so evident).

Reply:
Additionally - why are you subscribing at all in something that is called `TestPublisher`?

Reply:
We need to subscribe to the topic/subtopic. Then we start publishing messages on that topic/subtopic.